lib/services/apache.pm

```diff
@@ -0,0 +1,72 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Package for apache2 service tests
+#
+# Maintainer: Huajian Luo <hluo@suse.com>
+
+package services::apache;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use warnings;
+
+sub install_apache {
+    zypper_call("in apache2");
```

Comment:
better use single-tickets unless you rely on perl's variable expansion in strings.